emitter/jsonlog/emitter.go

```diff
@@ -126,6 +127,31 @@ func Emitter(w io.Writer, opt ...Option) alog.Emitter {
 			b.WriteString("}, ")
 		}
 
+		if len(e.STags) > 0 {
+			sTagPositions := make(map[string]int, len(e.STags))
+			for i, tag := range e.STags {
+				sTagPositions[tag.Key] = i
+			}
+			b.WriteString(`"sTags":{`)
+			for i, tag := range e.STags {
+				_, asStringTag := tagPositions[tag.Key]
+				if sTagPositions[tag.Key] != i || asStringTag {
+					continue
+				}
+
+				jsonString(b, tag.Key)
+				b.WriteByte(':')
+
+				marshalled, marshalErr := json.Marshal(tag.Val)
+				if marshalErr == nil {
+					b.Write(marshalled)
+				} else {
+					jsonString(b, "json marshal err: "+marshalErr.Error())
+				}
+				b.WriteString("}, ")
```

Comment:
Hmm, I feel like this `b.WriteString()` should be outside the loop.

Reply:
There were actually a couple bugs here, but I fixed them an updated the test.